pkg/ciContext/ciContext.go

```diff
@@ -70,22 +74,28 @@ func Extract() *CIContext {
 	}
 }
 
-func getCIName() string {
+func getCIData() (string, *CIMetadata) {
 	for env := range ciMap {
 		if isKeyInEnv(env) {
-			return env
+			return env, ciMap[env]
 		}
 	}
 
 	for _, key := range os.Environ() {
 		for prefix := range ciMapPrefix {
 			if strings.HasPrefix(key, prefix) {
-				return prefix
+				return prefix, ciMapPrefix[prefix]
 			}
 		}
 	}
 
-	return ""
+	for env := range ciFallbackEnvVar {
+		if isKeyInEnv(env) {
+			return env, ciFallbackEnvVar[env]
+		}
+	}
+
+	return "", nil
```

Comment:
just making sure, could this possibly lead us to `nil pointer` error in the future?

Reply:
No because who consumes it doesn't try to use any specific properties according my check